elcid/views.py

```diff
@@ -100,3 +104,105 @@ def get_context_data(self, *args, **kwargs):
+    then with the wards that are strings.
+    e.g. ICU
+    """
+
+    start = ward[:2].strip()
+    rest = ward[2:]
+    return (len(start), start, rest)
+
+
+class RenalHandover(LoginRequiredMixin, TemplateView):
+    template_name = "elcid/renal_handover.html"
+
+    def get_unit_ward(self, location):
+        if location.ward and location.bed:
+            return "{}/{}".format(
+                location.ward, location.bed
+            )
+
+    def get_context_data(self, *args, **kwargs):
+        """
+        An ordered dictionary of ward to patient
+
+        patient here is...
+        their name
+        their hospital number
+        their unit/ward
+        that episodes diagnosis
+        all of the clinical advice related to the patients infection service
+        """
+        ctx = super().get_context_data(*args, **kwargs)
+        episodes = Renal().get_queryset().order_by("-episode_id")
+        episodes = episodes.order_by("-start")
+        episodes = episodes.prefetch_related(
+            "location_set", "diagnosis_set"
+        )
+        by_ward = defaultdict(list)
+
+        # We don't want to show duplicate episodes in the case of
+        # multiple episodes of the same patient added to the renal list.
+        #
+        # So we just show the most recent episode per patient.
+        patient_ids = set()
+        for episode in episodes:
+            if episode.patient_id in patient_ids:
+                continue
+            patient_ids.add(episode.patient_id)
+            location = episode.location_set.all()[0]
+            diagnosis = ", ".join(
+                i.condition for i in episode.diagnosis_set.all()
+            )
+            demographics = models.Demographics.objects.filter(
+                patient__episode=episode
+            ).get()
+            microbiology_inputs = models.MicrobiologyInput.objects.filter(
+                episode__patient__episode=episode
+            ).order_by("-when")
+            primary_diagnosis = episode.primarydiagnosis_set.get().condition
+            by_ward[location.ward].append({
+                "name": demographics.name,
+                "hospital_number": demographics.hospital_number,
+                "unit_ward": self.get_unit_ward(location),
+                "primary_diagnosis": primary_diagnosis,
+                "diagnosis": diagnosis,
+                "lines": episode.line_set.all(),
+                "clinical_advices": microbiology_inputs,
+                "blood_culture_sets": episode.patient.bloodcultureset_set.all()
+            })
+
+        result = OrderedDict()
+        wards = by_ward.keys()
+        wards = sorted(wards, key=ward_sort_key)
+
+        other = None
+
+        for ward in wards:
+            if not ward:
+                other = by_ward[ward]
+            else:
+                result[ward] = by_ward[ward]
+
+        # add `other` last to the ordered dict
+        if other:
+            result["Other"] = other
+
```

Comment:
If I'm understanding you right, I think I might have spelled this as below and just not returned an ordered dict ;)
```python
if None in by_ward:
    by_ward["other"] = by_ward[None]
    del by_ward[None]

ward_names = sorted(by_ward.keys())
episodes_by_ward = [{'ward': w, 'episodes': by_ward[w]} for w in ward_names]
```

Reply:
implemented a version of this